### PR TITLE
Misc Cleanup of Server Side

### PR DESCRIPTION
**Pull request by TasoOneAsia**
Have not yet tested in game
[Misc Cleanup of Server Side](https://app.gitkraken.com/glo/card/d7897a50cff44eafb7749e378eb32b2e)

### CODE REVIEW

**Review by jfrader — APPROVED** [review]
Not tested. Lgtm

**Comment by RockySouthpaw** [comment]
This also fixes the SQL issue found in #155.

File: resources/server/sv_twitter.ts [hunk 1]
```diff
@@ -132,8 +133,12 @@ async function createTweet(identifier: string, tweet: NewTweet): Promise<Tweet>
     tweet.images,
     tweet.retweet,
   ]);
+  // This should not be an any type and instead should be
+  // a Tweet[] according to mysql2 documentation. But instead
```
Comment (erik-sn) [inline resources/server/sv_twitter.ts:137]: Are you sure on this? I am not sure this is the case.

`console.log(Object.keys(insertData));` is returning:

```
[
  "fieldCount",
  "affectedRows",
  "insertId",
  "info",
  "serverStatus",
  "warningStatus"
]
```

And this is the return value of the `await pool.query....`:

```
cfx> [
  {
    "fieldCount": 0,
    "affectedRows": 1,
    "insertId": 33,
    "info": "",
    "serverStatus": 2,
    "warningStatus": 0
  },
  null
]
```
and this is function is working as expected on latest master.

**Review by erik-sn — CHANGES_REQUESTED** [review]

File: resources/server/sv_twitter.ts (hunk 1 above)
Comment (TasoOneAsia) [reply resources/server/sv_twitter.ts:137]: Was a little out of it when I wrote that apparently as double checking now, we should use this type
```ts
declare interface ResultSetHeader {
    constructor: {
        name: 'ResultSetHeader'
    };
    affectedRows: number;
    fieldCount: number;
    info: string;
    insertId: number;
    serverStatus: number;
    warningStatus: number;
}
```

Comment (TasoOneAsia) [reply resources/server/sv_twitter.ts:137]: > and this is function is working as expected on latest master.

There is no change in functionality with this PR, just an added comment explaining the any

Comment (TasoOneAsia) [reply resources/server/sv_twitter.ts:138]: Fixed in 357acd4407b1368df55fd5a26a56dc849b8b8ec9

File: resources/server/sv_twitter.ts [hunk 2]
```diff
@@ -132,8 +133,12 @@ async function createTweet(identifier: string, tweet: NewTweet): Promise<Tweet>
...
+  // this exec promise returns the row itself, not an array of rows like it states
+  // Therefore type assertion doesn't work here for correctly typing this out.
   const insertData = <any>results;
-  return await getTweet(profile.id, insertData.insertId);
+  return await getTweet(profile.id, insertData.id);
```
Comment (erik-niehaus) [inline resources/server/sv_twitter.ts]: @TasoOneAsia this is the line I was mainly comenting on, I don't think `id` exists? `insertId` is changed here.

Comment (TasoOneAsia) [reply resources/server/sv_twitter.ts]: Fixed yea, I had forgotten I had changed that a couple days ago, should be resolved now with 357acd4

**Comment by TasoOneAsia** [comment]
Rebased to master

**Comment by RockySouthpaw** [comment]
> Rebased to master

Does this need to be rebased again before testing?

**Comment by TasoOneAsia** [comment]
> > Rebased to master
> 
> Does this need to be rebased again before testing?

Just rebased to master

**Comment by RockySouthpaw** [comment]
Some errors pertaining to messages if the person isn't online but i think it does this anyways. Messages is goofed as we know.

image


An error for calling someone who isn't online but this is normal. 

image

**Pull request merged**